src/Tests/Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests/ParameterCapturing/ObjectFormatting/Formatters/DebuggerDisplay/ExpressionBinderTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.ObjectFormatting.Formatters.DebuggerDisplay;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using System;
+using Xunit;
+using static Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.ObjectFormatting.Formatters.DebuggerDisplay.ExpressionBinder;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests.ParameterCapturing.ObjectFormatting.Formatters.DebuggerDisplay
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    public class ExpressionBinderTests
+    {
+        private sealed class DebuggerDisplayClass
+        {
+            public int Field = 10;
+            public static Uri StaticProperty { get; set; } = new Uri("http://www.bing.com/static");
+            public int Count { get; set; } = 10;
+
+            public Uri MyUri { get; }
+
+            public DebuggerDisplayClass(string uri)
+            {
+                RecursionProp = this;
+                MyUri = new Uri(uri);
+            }
+
+            public DebuggerDisplayClass RecursionProp { get; }
+            public DebuggerDisplayClass Recursion() => this;
+
+            public static void WithArgs(int i) { }
+
+            public string GetCountAsString()
+            {
+                return Count.ToString();
+            }
+
+            public void NoReturnType() => Count++;
+        }
+
+        [Theory]
+        [InlineData("GetCountAsString()", true, "10")]
+        [InlineData("DoesntExist()", false, null)]
+        [InlineData("WithArgs(Count)", false, null)]
+        [InlineData("Count", true, 10)]
+        [InlineData("Field", true, 10)]
+        [InlineData("NoReturnType()", true, null)]
+        // Bad chain
+        [InlineData(".", false, null)]
+        // Chained expression with implicit this type change
+        [InlineData("Recursion().RecursionProp.MyUri.Host", true, "www.bing.com")]
```

Comment:
IIRC example.com example.net example.org are all reserved names for purposes like this